Feature/app bar

baraa0abd opened the pull request:
![image](https://github.com/user-attachments/assets/9adca8f4-cfbc-45a6-9650-7eba14c095fb)
![image](https://github.com/user-attachments/assets/49f24649-19d9-48fa-aa7b-71b929f0c6fc)

baraa0abd committed:
feature/app-bar design (text style modification needed )

baraa0abd committed:
Create README.md

baraa0abd committed:
Update README.md

baraa0abd committed:
Refactor: Update AppBar styling and icon

This commit updates the styling of the `TasksAppBar` and `TudeeHeader` composables.

Key changes:
- `TasksAppBar`:
    - Changed background color from `Color(0xFF4AC0F2)` to `Color.White`.
- `TudeeHeader`:
    - Adjusted the order of modifiers for clarity.
- `CircularIconButton`:
    - Replaced `imageVector` with `painterResource(id = R.drawable.ic_back_button)` for the icon.
- `CombinedScreen`:
    - Removed the explicit background color, allowing the default or parent background to apply.

baraa0abd committed:
Merge remote-tracking branch 'origin/feature/app-bar' into feature/app-bar

baraa0abd committed:
Refactor TudeeHeader composable and add font families

This commit refactors the `TudeeHeader` composable in `AppBar.kt`.
It also introduces commented-out font family assignments for `cherryBombFamily` and `nunitoFamily`, suggesting an intention to use custom fonts.

Key changes:
- Removed an unnecessary import in `AppBar.kt`.
- Commented out the `fontFamily` assignments in the `TextStyle` for the header title and subtitle in `TudeeHeader`.
- Imported design system components, although they are not explicitly used in the changed code.

baraa0abd committed:
Update README.md

baraa0abd committed:
Update README.md

baraa0abd committed:
Update TudeeHeader font family

This commit updates the `TudeeHeader` composable to use `cherryBomOne` for the title and `nunito` for the subtitle.

baraa0abd committed:
Merge remote-tracking branch 'origin/feature/app-bar' into feature/app-bar

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+    Row(
+        modifier = modifier
+            .offset(8.dp, y = 25.dp)
+            .fillMaxWidth()
```
I really do not understand why you put offset here?

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+            .fillMaxWidth()
+            .background(
+                color = Color(0xFF4AC0F2),
+                shape = RoundedCornerShape(16.dp)
```
we must use the design systems colors Braa

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+                color = Color(0xFF4AC0F2),
+                shape = RoundedCornerShape(16.dp)
+            )
+            .wrapContentHeight()
```
I do not think that the background should be wrapContentHeight()

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+                shape = RoundedCornerShape(16.dp)
+            )
+            .wrapContentHeight()
+            .padding(0.dp),
```
what is the point of having 0 padding? you can delete it if you do not want use it

baraa0abd replied:
I forgot to delete it

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+                color = Color.White,
+                style = TextStyle(
+                    fontFamily = (cherryBomOne) ,
+                    fontWeight = FontWeight.Bold,
```
again, we must use the design systems themes here

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+                color = Color.White.copy(alpha = 0.9f),
+                style = TextStyle(
+                    fontWeight = FontWeight.W500,
+                    fontSize = 12.sp,
```
here again

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+            contentDescription = "Tudee avatar",
+            modifier = Modifier
+                .size(56.dp)
+                .clip(RoundedCornerShape(16.dp))
```
the image size in figma is 48

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+    contentDescription: String? = null,
+    backgroundColor: Color = Color.White,
+    iconTintColor: Color = Color.Black
+) {
```
nice work to parametrizing the button, but we wont need contentDescription to be parametrized, and for the icon use Painter because we all use it for consistency, and the color part again the same thing as before

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+    Row(
+        modifier = modifier
+            .offset(y = 20.dp)
+            .fillMaxWidth()
```
here again, I do not know why used offset

baraa0abd replied:
I use them to match the exact location of elements, just like they appear in Figma.

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+            .offset(y = 20.dp)
+            .fillMaxWidth()
+            .height(64.dp)
+            .background(Color.White, shape = RoundedCornerShape(16.dp))
```
the background should not have any Rounder Corners

abdulazizacc commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,144 @@
+            .background(Color.White, shape = RoundedCornerShape(16.dp))
+            .padding(horizontal = 16.dp, vertical = 12.dp),
+        horizontalArrangement = Arrangement.spacedBy(8.dp),
+        verticalAlignment = Alignment.CenterVertically
```
i do not think that the padding and Arrangement is right

baraa0abd replied:
Actually , I’ve implemented them correctly

abdulazizacc reviewed (changes requested):
Nice start Braa, but you must check comments!!

baraa0abd committed:
Refactor AppBar composables for improved styling and functionality

This commit introduces several changes to the AppBar composables:

**TudeeHeader:**
- Updates the background color to use `lightThemeColor.primaryColor.normal`.
- Modifies the "Tudee" text color to `lightThemeColor.primaryColor.variant`.
- Adjusts the size of the icon/avatar to `48.dp`.
- Removes `wrapContentHeight()` and `padding(0.dp)` from the main Row modifier.

**CircularIconButton:**
- Changes the `icon` parameter type from `ImageVector` to `Painter` to allow for drawable resources.

**TopAppBar:**
- Changes the background shape of the `TopAppBar` from `RoundedCornerShape(16.dp)` to a simple `Color.White` background.
- Updates the back button icon in `TopAppBar` to use `painterResource(id = R.drawable.ic_back_button)`.

hagarabobakr commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,146 @@
+            modifier = Modifier,
+            text = "Tasks",
+            style = MaterialTheme.typography.titleLarge,
+            color = Color.Black
```
Use the design system's colors, please.

hagarabobakr reviewed (commented):
Great Job Braa ❤️

baraa0abd committed:
Update AppBar text and icon colors for light theme

This commit updates the `AppBar` and `AppBarWithBackButton` composables to use colors from the `lightThemeColor` object.

- In `AppBar`:
    - The subtitle text color is changed from `Color.White.copy(alpha = 0.9f)` to `lightThemeColor.textColor.onPrimary.copy(alpha = 0.9f)`.
- In `AppBarActionButton`:
    - The default `backgroundColor` is changed from `Color.White` to `lightThemeColor.textColor.onPrimary`.
    - The default `iconTintColor` is changed from `Color.Black` to `lightThemeColor.status.emojiTint`.
- In `AppBarWithBackButton`:
    - The back button icon tint color is changed from `Color.Black` to `lightThemeColor.status.emojiTint`.
    - The title text color is changed from `Color.Black` to `lightThemeColor.status.emojiTint`.

hagarabobakr commented on app/src/main/java/com/baghdad/tudee/ui/composable/AppBar.kt:
```diff
@@ -0,0 +1,146 @@
+        Column {
+            Text(
+                text = "Tudee",
+                color =  lightThemeColor.primaryColor.variant,
```
Use general color theme

baraa0abd committed:
Refactor AppBar with Theme and TextStyle

This commit refactors the `AppBar.kt` to utilize the new `Theme` and `tudeeTextStyle` objects for consistent styling.

Key changes:
- Replaced direct color and text style definitions with references from `Theme.color` and `tudeeTextStyle`.
- Updated `TudeeHeader` to use `Theme.color.primaryColor.normal` for background and `tudeeTextStyle.logo` and `tudeeTextStyle.label.small` for text styling.
- Updated `EmojiButton` to use `Theme.color.textColor.onPrimary` for background and `Theme.color.status.emojiTint` for icon tint.
- Updated `AppBarBack` to use `Theme.color.surfaceColor.surfaceHigh` for background and `Theme.color.status.emojiTint` for icon and text color.
- Removed unused imports for `Icons`, `ArrowBack`, `FontFamily`, `FontWeight`, `sp`, `nunito`, `cherryBomOne`, and `lightThemeColor`.

baraa0abd committed:
Merge remote-tracking branch 'origin/develop' into feature/app-bar

baraa0abd committed:
refactor the AppBar

Hasa9nAli committed:
Merge branch 'develop' into feature/app-bar

abdulazizacc reviewed (approved):
nice way improving ur code

Arzo-zi reviewed (approved):
great job
